### DISCUSSION TITLE
🦊 - Announcing Video DownloadHelper V9-Beta

### DISCUSSION THREAD

**Discussion by paulrouget** [aclap-dev/video-downloadhelper#466] [Announcements]
You can get it here: https://downloadhelper.net/beta

… or if you're already on the Beta channel, it will autoupdate in the coming 24 hours.

### New Design

We have updated the main user interface with a more modern look, support for black/light theme switching, and accessibility navigation. **This new look is optional, you can always re-enable the legacy look in the settings**.

### Sidebar Support

We can now dock Video DownloadHelper into the sidebar for a live view of available downloads, current downloads, and downloaded videos.

---

Thank you for testing the beta ❤️ And please don't forget to report any bugs / feedback you might have!

<img width="450" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/0aa9ee77-c1f9-45dc-ba58-90807274cd34">

**Comment by ToasterUwU** [comment]
Looks good, but for some reason i still see the old UI, even after installing the Beta with the button.

I dont know which version is the new one, but the one i get by clicking the Beta button is the one below, it looks like its too old to be the right one, but thats all i get by clicking the beta button, maybe the link isnt updated properly yet?

version: 8.2.2.8a1
channel: beta
build date: 2024-03-14
build options: linuxlic: false, noyt: false, browser: firefox
platform: x86-64 linux

Edit: Just saw it says "V9" in the title, so i was right. The link gives me the 8.2.2.8a1 version which is old

**Reply by GrampaWildWilly** [reply]
You probably need to follow the instructions over here: #391.  You have to hunt a bit for the instructions but they're in there.  It seems the Wiki finally got updated to what was, until yesterday, the latest beta.  But now, the newest VDH beta for Firefox is version 9, & the only way to get it is as I describe over there in #391.  I imagine the Wiki will eventually get fixed.  But for now, use the browser's own tools.

**Comment by paulrouget** [comment]
@ToasterUwU sorry, it's a caching issue. Can you retry the link?

**Reply by ToasterUwU** [reply]
I assumed it was something like that, works now

**Comment by jcfrog** [comment]
@ToasterUwU please try to update the addon, it should update with v9


<img width="1222" alt="image" src="https://github.com/aclap-dev/video-downloadhelper/assets/368367/cc4cdfbe-bff4-4e86-86d7-a86d47657173">

**Reply by ToasterUwU** [reply]
That also worked, thanks

**Comment by GrampaWildWilly** [comment]
I must say that the greatest improvement in this version is the ability to operate VDH strictly from the keyboard.  Mouse clicks out the wazoo are no longer necessary.

Also, I highly recommend the setting to "Make the addon popup larger."  I would have chosen the words "Make the VDH menu wider."  But let's not quibble.  I'm hoping we get a new setting that allows us to choose just how wide we want the menu to be.

Speaking of settings, there is no separate dialog window like before.  Everything is right there in the VDH menu.  Plus you don't need to remember to click a Save button.  Your changes are immediate.

Another separate window that has gone away is the download progress status window from the blue dot.  The VDH menu itself will show progress while a download is in flight.  It's not perfect yet, but it is much more convenient than before.

You used to access a secondary menu by hovering over a variant to make an arrow appear, then clicking the arrow.  This is completely different now.  To the right of the word "Download" there is a small control that drops down a menu.

We used to have Quick Download & Download.  Now we have Download & Download As.  We do still have a way to set a default action.  It works a lot like the old way.  Experiment.  You'll figure it out.

Although it is possible to set the default download directory in the VDH menu, & your setting is correctly remembered, it isn't yet used properly.  When the file selection dialog opens, it puts you into a default directory for your user.  You have to manually navigate to your desired directory each time.  A minor annoyance that I'm sure Paul will fix in due time.  My vote is that it get fixed right away but I'm just one voter.  And speaking of the file selection dialog, it is the native file picker.  You no longer have to explicitly set something to make that happen.  It's just how it works now, no option available.  Just as well.  I always thought it was a lot of wasted effort for VDH to have a custom file selection dialog when the operating system already supplies one.

**Reply by paulrouget** [reply]
> the ability to operate VDH strictly from the keyboard

I have a plan to actually make that even better :)

> Although it is possible to set the default download directory in the VDH menu, & your setting is correctly remembered, it isn't yet used properly.

This is now high in my todolist. Hope to see that happening in the coming days!

**Reply by GrampaWildWilly** [reply]
I live for mouselessness.

And that last directory thing can get truly annoying if you do a lot of downloads.  Fortunately, that's not my daily habit.  But I know there's others who will welcome it even more eagerly than I will.

**Comment by ToasterUwU** [comment]
Ok. Now that the new version actually got downloaded, its great. On first look there are no issues and everything looks great. Dark mode is good for my eyes, all the menus and buttons are super easy to read and way more intuitive, the sidebar thing makes this so much easier to handle, and i love that the selection of a download directory is the native filepicker now instead of the custom one.

All in all its great, with all these usability improvements and the better design, its now not only the most capable download extension i know, but also the best looking and easiest to use.
I will keep my eyes open and report back if i find any issues.

**Reply by paulrouget** [reply]
Thank you so much ❤️

Let us know if you find any issues :)

**Comment by nollinvoyd** [comment]
Just installed.  A few initial impressions.   Not criticisms!  Much love for VDH.
  
1. No biggie, but some settings reverted to default, i.e. the download folder

2. No badges?

3. Can I set it to open in the sidebar by default?

4. My biggest issue, with respect to the previous post, is that for me, the font in some areas is tiny.  In others, i.e. the file titles, in your sample posted image it has bold type.  Mine does not.  I don't exactly know why, maybe my configuration? That being said, why not have the bold font at least in any area where interface space would not somehow be compromised, if in fact that would be a reason to limit it in certain areas.  I wear glasses, so I'm squinting at some things.  Would it be possible to add controls for this? That image I posted doesn't accurately represent how small those things appear to me in real time.

![vdh](https://github.com/aclap-dev/video-downloadhelper/assets/20844322/9c0a06f4-8a32-42c8-9833-855d52a893a1)

**Reply by paulrouget** [reply]
> No biggie, but some settings reverted to default, i.e. the download folder

I'll fix that.

> No badges?

You mean the numbers at the bottom of the popup in the previous version? Can you explain how they would be helpful for you in v9.

> Can I set it to open in the sidebar by default?

If you click on the sidebar button (at the bottom) then it will always open in the sidebar.

> the font in some areas is tiny.

Noted! I'll figure something out!

**Reply by GrampaWildWilly** [reply]
I'm running Windows 11 & the font handling in the operating system is not the best.  I found a tutorial on elevenforum that explains how to set the global system font.  Problem is, it doesn't actually make the change globally.  Many things obey my change for the system font.  For example on this web page, the text of posts is in my chosen font.  Ditto for my own posts while I'm typing them, ditto for everything within the body of the web page.  But the URL in the URL bar is in some ugly Microsoft default font.  Ditto for the menu bar, ditto for every entry in any menu, ditto for the window title in the top border.  In the Firefox settings dialog, everything is in the Microsoft font, not my chosen changed font.  So any issues in the VDH dialogs is probably going to be a royal pain to deal with.  Plus I'll bet whatever makes things nice on Windows won't be applicable on Linux nor Mac.  I totally sympathize with you about glasses.  My eyesight could be better, too.  So I'm very picky about fonts, font sizes, colors, all that sort of thing.  I don't envy Paul trying to get this set right.  I'm quite prepared to hear him say at a certain point that it's the best he can do & the rest is just not within his control.

**Reply by GrampaWildWilly** [reply]
Here's what I'm talking about:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/bdae4cd4-6c90-46ec-a206-a4657dbea661)

If you're not used to looking for such things, it may seem hard to tell the difference.  But my font has serifs & the font I don't want has no serifs.  In a past life I learned that serifs make things easier to read.  Sans serif fonts are harder to read.  It amazes me that Microsoft has chosen a sans serif font as the initial system default.  It is truly ugly.  I want to punch my monitor when I see any large amount of sans serif text.  I suspect somebody decided it's more sleek, more modern looking, to dispense with serifs.  It's scientifically a bad idea.  Of course, nobody cares about science.

**Reply by paulrouget** [reply]
I'm using the system font when possible.

So we have different fonts on Windows, Mac and Linux.

On Mac (where we took the screenshots), the foot is bolder than Windows.

I can force a font (looking at _Fira_ - and no, sorry, no serif, it's not 1980 anymore 😆) then everybody has the exact same font.

Or force bold font on Windows only.

Experimenting…

**Reply by GrampaWildWilly** [reply]
System font when possible sounds like a very good idea.  Everybody with the same font sounds like not a very good idea.  I think forced bold font is another not very good idea.  And inability to have serif fonts is another really quite bad idea.  It has nothing to do with 1980.  Serif fonts are simply easier to read.  I don't believe ease of reading is something that ever gets outdated.  The VDH menu seems to have picked up my choice for system font so I think it's fine the way it is.  I'm not so sure about the all caps italics, but that's getting a bit fine.  I suspect the all caps is the result of something on the web page.  The italics might be something you're doing to indicate something.  That the item hasn't been downloaded yet?  You change the Download button to Play after a download, so I don't think the italics are really necessary.

**Reply by GrampaWildWilly** [reply]
Just noticed something nearly insignificant.  There is no tooltip on the Settings gear in the bottom of the VDH menu.  The other 4 icons down there have tooltips.  Can't imagine why you'd discriminate against Settings . . .

**Comment by GrampaWildWilly** [comment]
By the way, we're up to 9.0.1.8a1, folks.

**Comment by paulrouget** [comment]
I've pushed 9.0.1.9a1, which, hopefully, make the font look slightly better on Windows.

Basically, font is a little bolder. And title gets bold if the large UI option is checked.

**Reply by nollinvoyd** [reply]
Thanks.  Bold fonts, and maybe I didn't realize before, but now opens in sidebar, if I wish, by default.

Let me clear up a misconception I apparently created in my previous post.  The vision challenges I mentioned weren't referring to my system font versus your choice of font type in VDH (which, for me, display the same, BTW), just the bold aspect and font size in the VDH interface.

I was hoping for some flexibility within in the settings for manually manipulating font size/display, just as we find in some other apps.  

Under the current circumstances, I do prefer the dark theme,  Better contrast, but if I could manipulate the font size, I'd probably return to "system/light"

I also would not have chosen italics for the titles.  That's just me about about anything I could personally achieve to further compliment age-related viewing comfort level challenges, even if it's only a bit.  Allows me to better and more efficiently focus on my tasks at hand without the built-in, subconscious distractions.

Is "Hits" a reference to Vision & Audio or the available list of downloads? That could be a bit more prominent size-wise.

**Reply by GrampaWildWilly** [reply]
I do not like bold italic all caps.  There you go.  Different people have different tastes.  I would prefer, like @nollinvoyd above, to have control of the choice of font & font size.  I would probably choose to use the system default.  And titles should be mixed case, not italic, & not bold.  At least when you launch a download & get into the file selection dialog, the file name is in mixed case.  At least bold & italic are not issues in the file selection dialog because you can't make a file name either bold or italic.  But mixed case in file names is perfectly acceptable & I'm glad to see that's what VDH is offering.  But the menu should faithfully report the file name that you will ultimately get when you launch a download.  That means mixed case, NOT bold, NOT italic.

**Comment by paulrouget** [comment]
FYI it is now available for Chrome Beta too.

**Comment by smthngelse** [comment]
A couple of quick notes, apologies if I missed a reference to them somewhere...

The new beta on Chrome now requests an additional permission, "Open existing files" or something like that.  Sorry, I thought it would show me the error again when I checked the settings so I didn't screen cap it or anything.  A mention in the release announcement about why it needs an additional permission would be good.

Also, older version shows bitrate if there are multiple streams available...  this one doesn't.  For example, one site I use sometimes offers 1080p in multiple bitrates and I like to make sure I'm picking the highest.  I may pick a lower bitrate on 4k streams though.  The current version only show that there are two 1080p streams, not which is which.

Thanks for keeping the legacy view available though...  I will definitely be using that for some sites that i pick different resolutions on each lesson.  :)

**Comment by paulrouget** [comment]
> A mention in the release announcement about why it needs an additional permission would be good.

Good catch! Indeed with need that to open certain downloaded files.

> multiple bitrates

It should show the bitrate if there's multiple bitrates available for similar quality.

**Reply by smthngelse** [reply]
The new version does show multiple streams, but doesn't indicate that there are duplicates because there are different rates.  Here's the original downloader clip:
![image](https://github.com/aclap-dev/video-downloadhelper/assets/28632685/cd4b8a8b-1ad1-4aa5-b361-f78420969a52)

Here's the new one after clicking on the drop-down arrow for resolution:
![image](https://github.com/aclap-dev/video-downloadhelper/assets/28632685/637a1492-64ba-485c-9d0e-2271dc04e08c)

It shows two 720 streams, but no indication that they're different bitrates.  

Let me know if those two clips don't show, I'll pop them up somewhere else if needed.  TIA!

**Reply by paulrouget** [reply]
Ah! You're correct. I thought I fixed that. I'll address this issue in an upcoming update.

**Reply by paulrouget** [reply]
This will be fixed in 9.0.1.10.

**Comment by Mattmjs** [comment]
Is there anything which will be deprecated going forward in this new interface. The old interface still has _analyze page_ & 
_aggregate local audio and video files_  buttons but not present in new interface. The _analyze page_ is still broken for those who use it.

**Reply by paulrouget** [reply]
> analyze page

That's something that has been broken since January iirc. I haven't had a chance to fix it yet.

> aggregate local audio and video files buttons

I'm not sure I want to port these features from V8 to V9. They are barely used ever. And make the code lot more complex. Still thinking about it.

**Reply by GrampaWildWilly** [reply]
It's so easy to merge video & audio using ffmpeg directly that it's hardly worth it to have it as a feature in VDH.  And ffmpeg is already available to everybody because it's part of the CoApp.

ffmpeg -i "video track file name" -i "audio track file name" -codec: copy "merged video with audio output file name"

Be sure you code the quotation marks.  They are important.  The file names would normally include complete file specifications of the form x:\dir\dir\dir\file name.ext.

**Comment by paulrouget** [comment]
Version 9.0.1.10 published with:

- bitrate added to variant menu /cc @smthngelse
- bunch of locales updated
- migration from some v8 prefs into v9
- stop using HOME folder as DownloadAs initial directory, but whatever was set a default download directory in the prefs /cc @GrampaWildWilly

**Reply by GrampaWildWilly** [reply]
Excellent!

**Comment by DavidBE-UK** [comment]
Just checking - I'm still not seeing a version of the old 'Remember last directory' download location option, or have I missed something? That is THE most important missing feature for me.

**Reply by paulrouget** [reply]
Right - for now it's using the default download folder in the Settings.

But I'll add re-add the "remember last directory" option in the coming day! Sorry 🫤

**Reply by DavidBE-UK** [reply]
Thanks. :)

**Comment by bughit** [comment]
> Good catch! Indeed with need that to open certain downloaded files.

@paulrouget Please clarify, which downloaded files need to be opened by v9 and why?

**Comment by paulrouget** [comment]
> > Good catch! Indeed with need that to open certain downloaded files.
> 
> @paulrouget Please clarify, which downloaded files need to be opened by v9 and why?

If you download a file without the coapp (this one for example: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video ) we need a way to play it or open its directory (the green button after the download has finished).

Usually it's done by the coapp, but if there's no coapp, we need to use Firefox downloads.open API.

**Comment by GrampaWildWilly** [comment]
Should we be expecting a 9.0.1.14a1 beta or something like that to match what's gone in the standard release channel?

**Reply by paulrouget** [reply]
Mozilla is slow to sign - doing my best to keep beta and stable in sync.

**Reply by GrampaWildWilly** [reply]
Oh.  Right.  I knew that.  Patience is in order . . .
